rfc/0054-sdk3-management-apis.md

```diff
@@ -582,6 +666,40 @@ void BuildDeferredIndexes(string bucketName, [options])
 
   * `Timeout` or `timeoutMillis` (`int`/`duration`) - the time allowed for the operation to be terminated. This is controlled by the client.
 
+  * `CollectionName` - the name of the collection to restrict indexes to.
+  * `ScopeName` - the name of the scope to restrict indexes to.
+    * If either `CollectionName` or `ScopeName` are set then both *must* be set.
+
+### N1QL
+
+This is done by performing 2 querys; one to fetch the deferred indexes to build and the other to build the deferred
+indexes.
```

Comment:
Sergey and I chatted and came to the conclusion that the query for the deferred index names could be replaced by a call to getAllIndexes (post-filtered by status = "deferred"), as long as the call to getAllIndexes is limited to the target collection. Using getAllIndexes is less efficient, but potentially simpler to code and maintain. Apart from efficiency, is there another reason to use a special query to find the deferred index names?